### ISSUE TITLE
False positive - E1101: Instance of 'tuple' has no member

### ISSUE THREAD

**Issue by kfiri** [PyCQA/pylint#2964]
### Steps to reproduce
1. Just copy the following code to `nodes.py`
```python
class Node:
    def __init__(self, name, path=()):
        """
        Inisialize self with "name" string and the tuple "path" of its parents.
        "self" is added to the tuple as its last item.
        """
        self.__name = name
        self.__path = path + (self,)

    def get_full_name(self):
        """
        When iterating over the items in "self.__path", pylint
        detects the items in path as tuples instead of instances of Node.
        """
        # "node.__name" is marked as an error for pylint(no-member) -
        # nodes.py:17:24: E1101: Instance of 'tuple' has no '__name' member (no-member)
        return '.'.join(node.__name for node in self.__path)
```
2. Run `pylint ./nodes.py`
3. The error `nodes.py:17:24: E1101: Instance of 'tuple' has no '__name' member (no-member)` should not appear.

### Current behavior

The error `nodes.py:17:24: E1101: Instance of 'tuple' has no '__name' member (no-member)` appears for no reason.

**Note** that this error would not occur if "(self,)" is not added to self.__path on the same line:
```python
# If this was written in the __init__ method:
self.__path = path
self.__path += (self,)
# The line '.'.join(node.__name for node in self.__path) in get_full_name
# would not cause the error to appear.
```

### Expected behavior

The error `nodes.py:17:24: E1101: Instance of 'tuple' has no '__name' member (no-member)` should not appear for the tuple of `Node`s.
pylint should treat the sum of 2 tuples as a tuple and not as a sequence of tuples.

### pylint --version output
pylint 2.3.1
astroid 2.2.5
Python 3.7.3 (v3.7.3:ef4ec6ed12, Mar 25 2019, 22:22:05) [MSC v.1916 64 bit (AMD64)]

**Comment by PCManticore** [comment]
Thanks for the report!

**Comment by PCManticore** [comment]
Thanks for the report!

**Comment by mbyrnepr2** [comment]
Testing on the latest astroid version(2.13.0) & this one doesn't occur anymore. Please re-open if the issue persists.
